### PR TITLE
Avoid classical for if for...of when looping all object of an Array-like object (part #4)

### PR DESCRIPTION
**Pull request by teoli2003**
Classical `for` loops are error-prone (setting the right condition, not mixing indices…) and difficult to read when used to loop on all elements of an Array-like structure.

This PR changes some occurrences to use the modern `for…of`, or `.forEach()` in some occurrences.

Note that:
- I use `for…of` if I don't need the index
- I use `.forEach()` when I do, or on an object where `for…of` may not work.
- There were a couple of cases where the loop copied a _typed array into a new one; I used `UintXYArray.from()` instead.

### CODE REVIEW

File: files/en-us/web/api/cssstyledeclaration/setproperty/index.md [hunk 1]
```diff
@@ -160,13 +160,7 @@ function randomColor() {
 }
 
 const stylesheet = document.styleSheets[1];
-let boxParaRule;
-
-for (let i = 0; i < stylesheet.cssRules.length; i++) {
-  if (stylesheet.cssRules[i].selectorText === '.box p') {
-    boxParaRule = stylesheet.cssRules[i];
-  }
-}
```
Comment (Josh-Cena) [inline files/en-us/web/api/cssstyledeclaration/setproperty/index.md:169]: This is selecting a CSS rule, not a DOM element

File: files/en-us/web/api/element/getelementsbyclassname/index.md [hunk 2]
```diff
@@ -117,17 +117,12 @@ then become `item(0)`.
...
+We can also use methods of {{jsxref("Array.prototype")}} on any {{domxref("HTMLCollection")}} by transforming the {{domxref("HTMLCollection")}} in an array.
+Here we'll find all {{HTMLElement("div")}} elements that have a class of `test`:
 
 ```js
 const testElements = document.getElementsByClassName('test');
-const testDivs = Array.prototype.filter.call(
-  testElements,
-  (testElement) => testElement.nodeName === 'DIV',
-);
+const testDivs = [...testElements].filter((testElement) => testElement.nodeName === 'DIV');
```
Comment (Josh-Cena) [inline files/en-us/web/api/element/getelementsbyclassname/index.md:125]: FWIW, I prefer the code before. WDYT?

File: files/en-us/web/api/datatransferitemlist/clear/index.md [hunk 3]
```diff
@@ -126,8 +123,6 @@ function dragend_handler(ev) {
 
 {{EmbedLiveSample('Examples', 400, 300)}}
 
-{{LiveSampleLink('Examples','Result link')}}
```
Comment (Josh-Cena) [inline files/en-us/web/api/datatransferitemlist/clear/index.md:129]: This is expected?

File: files/en-us/web/api/datatransferitemlist/clear/index.md [hunk 4]
```diff
@@ -68,55 +68,52 @@ This example shows the use of the `clear()` method.
...
   console.log("dragStart");
+
   // Add this element's id to the drag payload so the drop handler will
   // know which element to add to its tree
-  let dataList = ev.dataTransfer.items;
+  const dataList = ev.dataTransfer.items;
   dataList.add(ev.target.id, "text/plain");
-  // Add some other items to the drag payload
-  dataList.add("<p>Paragraph…</p>", "text/html");
-  dataList.add("http://www.example.org","text/uri-list");
```
Comment (Josh-Cena) [inline files/en-us/web/api/datatransferitemlist/clear/index.md:79]: This is expected?

File: files/en-us/web/api/datatransferitemlist/clear/index.md (hunk 3 above)
Comment (teoli2003) [reply files/en-us/web/api/datatransferitemlist/clear/index.md:129]: Yes, this only links to a full page of the EmbedLiveSample above that is working well.

File: files/en-us/web/api/datatransferitemlist/clear/index.md (hunk 4 above)
Comment (teoli2003) [reply files/en-us/web/api/datatransferitemlist/clear/index.md:79]: Yes, these were added but useless, and obfuscating the output.

File: files/en-us/web/api/element/getelementsbyclassname/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/api/element/getelementsbyclassname/index.md:125]: I'm not a fan of calling prototype functions on different objects. I find this complex to understand, but that's a matter of taste.

Let's have opinions from people outside the modernization project. They likely have fresh eyes, @elchi3, @wamberg.

(I have no strong feelings here)

Comment (Elchi3) [reply files/en-us/web/api/element/getelementsbyclassname/index.md:125]: I guess the thing here is that one way or another you need to convert from array-like to `Array`. I'm not sure what communicates that most explicitly. 
The `[...testElements]` syntax is new and you have to know what is going on. So, if you use this, maybe add a comment?

The `Array.prototype.filter.call` variant reads a bit old and strange but in the text above it says "We can also use methods of {{jsxref("Array.prototype")}} on any..." and now you now longer explicitly call such a method (and rather use syntactic sugar).

I believe in the context of this whole section "Filtering the results using array methods" the `Array.prototype.filter.call` variant is probably better. Open  to suggestions that change the whole section, though, so it makes more sense overall.

File: files/en-us/web/api/cssstyledeclaration/setproperty/index.md [hunk 5]
```diff
@@ -160,13 +160,7 @@ function randomColor() {
...
+const boxParaRule = document.querySelector(".box p");
```
[the remark this answers is not in the document]
Comment (Josh-Cena) [reply files/en-us/web/api/cssstyledeclaration/setproperty/index.md:163]: Right—already expressed the same concern in https://github.com/mdn/content/pull/19544#discussion_r945276608

File: files/en-us/web/api/element/getelementsbyclassname/index.md (hunk 2 above)
Comment (teoli2003) [reply files/en-us/web/api/element/getelementsbyclassname/index.md:125]: I'm reverting this. Maybe in a year or two, `[...testElements]` will be idiomatic, but then we will revisit all cases of `Array.prototype.xyz.call` at that time.

File: files/en-us/web/api/cssstyledeclaration/setproperty/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/cssstyledeclaration/setproperty/index.md:163]: You are right: I'm replacing this with `for...of` (Note that `forEach()` wouldn't work, it doesn't exist on a `CSSRuleList`.

Commit incoming.

[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/cssstyledeclaration/setproperty/index.md:163]: Duplicate of above (commit incoming), I'm resolving this one.

**Review by rubiesonthesky — COMMENTED** [review]
These look good to me.

**Review by Josh-Cena — APPROVED** [review]
Thanks, this LGTM

**Pull request merged**